public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/index.tsx

```diff
@@ -5,29 +5,40 @@ import { css } from '@emotion/css';
 import { GrafanaTheme2 } from '@grafana/data';
 import { Button, useStyles2 } from '@grafana/ui';
 import ResourcePickerData from '../../resourcePicker/resourcePickerData';
-import { produce } from 'immer';
 import { Space } from '../Space';
-import { findRow, parseResourceURI } from './utils';
+import { addResources, findRow, parseResourceURI } from './utils';
 
 interface ResourcePickerProps {
-  resourcePickerData: Pick<ResourcePickerData, 'getResourcePickerData' | 'getResourcesForResourceGroup'>;
+  resourcePickerData: ResourcePickerData;
   resourceURI: string | undefined;
+  templateVariables: string[];
 
   onApply: (resourceURI: string | undefined) => void;
   onCancel: () => void;
 }
 
-const ResourcePicker = ({ resourcePickerData, resourceURI, onApply, onCancel }: ResourcePickerProps) => {
+const ResourcePicker = ({
+  resourcePickerData,
+  resourceURI,
+  templateVariables,
+  onApply,
+  onCancel,
+}: ResourcePickerProps) => {
   const styles = useStyles2(getStyles);
 
-  const [rows, setRows] = useState<ResourceRowGroup>([]);
+  const [azureRows, setAzureRows] = useState<ResourceRowGroup>([]);
   const [internalSelected, setInternalSelected] = useState<string | undefined>(resourceURI);
 
   // Sync the resourceURI prop to internal state
   useEffect(() => {
     setInternalSelected(resourceURI);
   }, [resourceURI]);
 
+  const rows = useMemo(() => {
+    const templateVariableRow = resourcePickerData.transformVariablesToRow(templateVariables);
+    return [...azureRows, templateVariableRow];
```

Comment:
What if there are no variables? I guess we don't want to add a row for template variables then.